codegen-server-test/build.gradle.kts

```diff
@@ -38,6 +38,7 @@ dependencies {
 val allCodegenTests = "../codegen-core/common-test-models".let { commonModels ->
     listOf(
         CodegenTest("crate#Config", "naming_test_ops", imports = listOf("$commonModels/naming-obstacle-course-ops.smithy")),
+        CodegenTest("casing#ACRONYMInside_Service", "naming_test_casing", imports = listOf("$commonModels/naming-obstacle-course-casing.smithy")),
```

Comment:
There's no harm in adding this to `codegen-client` too.

Reply:
https://github.com/awslabs/smithy-rs/pull/2349/files#diff-0e456b9761f195113cafd0d3a68068a9bd6dba3a0d4d37e8bf93d38209d36e50R82-R86